src/rgw/driver/posix/rgw_sal_posix.cc

```diff
@@ -1639,6 +1723,51 @@ int POSIXObject::POSIXReadOp::prepare(optional_yield y, const DoutPrefixProvider
     }
   }
 
+  if (!rgw::sal::get_attr(source->get_attrs(), RGW_ATTR_ETAG, etag_bl)) {
+    return -EINVAL;
+  }
+
+#if 0 // WIP
```

Comment:
@dang what's the purpose of this section?

Reply:
Work in progress for if-modified.  It didn't build yet, so I commented it out.